### PR TITLE
Chore: update azure grafana sdk (go mod)

### PR DESCRIPTION
**Pull request by ryantxu**
While trying to link to a new plugin SDK build, I ran into weird go.mod resolution issues, that seemed super similar to things @sdboyer discussed in https://github.com/grafana/grafana/pull/60207#issuecomment-1347703083

I admit I do not really understand the cascading dependencies 😬 -- just that trying to `go get <new grafana sdk>` ended up with breaking changes in the azure sdks that the 1.5.0 SDK update seems to fix 🤷 

If this upgrade is not desired, we can likely find another way

### CODE REVIEW

File: pkg/tsdb/azuremonitor/credentials_test.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func TestCredentials_getAzureCloud(t *testing.T) {
 	t.Run("when auth type is managed identity", func(t *testing.T) {
 		jsonData := simplejson.NewFromAny(map[string]interface{}{
 			"azureAuthType": azcredentials.AzureAuthManagedIdentity,
-			"cloudName":     azureMonitorGermany,
+			"cloudName":     azureMonitorUSGovernment,
```
Comment (ryantxu) [inline pkg/tsdb/azuremonitor/credentials_test.go:91]: https://github.com/grafana/grafana-azure-sdk-go/pull/14

File: pkg/tsdb/azuremonitor/azuremonitor_test.go [hunk 2]
```diff
@@ -52,40 +52,40 @@ func TestNewInstanceSettings(t *testing.T) {
...
-						},
-					},
-				},
-				DatasourceID:            50,
-				DecryptedSecureJSONData: map[string]string{"clientSecret": "secret"},
-				Services:                map[string]types.DatasourceService{},
-			},
-			Err: require.NoError,
-		},
+		// {
```
Comment (ryantxu) [inline pkg/tsdb/azuremonitor/azuremonitor_test.go:55]: any ideas here?

File: go.mod [hunk 3]
```diff
@@ -58,7 +58,7 @@ require (
 	github.com/gorilla/websocket v1.5.0
 	github.com/grafana/cuetsy v0.1.1
 	github.com/grafana/grafana-aws-sdk v0.11.0
-	github.com/grafana/grafana-azure-sdk-go v1.3.1
+	github.com/grafana/grafana-azure-sdk-go v1.5.0
```
Comment (ryantxu) [inline go.mod:61]: ^^ the rest fall from `go mod tidy`

**Comment by andresmgot** [comment]
hi @ryantxu, sorry for not fixing this earlier. I have created https://github.com/grafana/grafana/pull/60372 to address this with minimal changes.

**Comment by ryantxu** [comment]
Closing in favor of https://github.com/grafana/grafana/pull/60372

**Pull request closed without merging**